scaling_performance/host_practices.adoc

```diff
@@ -57,42 +57,94 @@ include writing a large amount of log data.
 etcd is a distributed key-value store that {product-title} uses for
 configuration.
 
+Version 3 of etcd brings with it important scalability and performance improvements that reduce CPU, memory, network and disk requirements for any size cluster.  OpenShift 3.5 introduced etcd version 3.1.
+
+Version 3 of etcd implements a backwards compatible storage API that facilitates a two-step migration of the on-disk etcd database.  For migration purposes, the storage mode used by etcd 3.1 in OpenShift 3.5 remained in v2 mode.  As of OpenShift 3.6, new installs will use storage mode v3.  Upgrades from previous versions of OpenShift will NOT automatically migrate data from v2 to v3.  This must be done using supplied playbooks following the documented process.
+
+In addition to changing the storage mode for new installs to v3, OpenShift 3.6 also begins enforcing "quorum reads" for all OpenShift types.  This is done to ensure that queries against etcd do not return stale data.  In single-node etcd clusters, stale data is not a concern.  In highly available etcd deployments typically found in production clusters, quorum reads ensure valid query results.  Quorum reads force an fsync to guarantee all nodes are in sync.  This behavior is part of the raft algorithm design.
+
+It is important to note that OpenShift uses etcd for storing additional information beyone what Kubernetes itself requires.  For example, OpenShift stores information about images, builds, RBAC, secrets (amongst others) in etcd as is required by features that OpenShift adds on top of Kubernetes.  Ultimately what this means is that guidance around performance and sizing for etcd hosts will differ from Kubernetes and other recommendations in salient ways.  Red Hat tests etcd scalability and performance with the OpenShift use-case and parameters in mind to generate the most accurate recommendations.
+
+Performance improvements have been quantified using a 300 node OpenShift 3.6 cluster using the https://github.com/openshift/svt/tree/master/openshift_scalability[cluster-loader] utility.
+
+Comparing etcd 3.1 (storage mode v2) versus etcd 3.1 (storage mode v3), clear improvements are identified.  Based on extensive test results, Red Hat strongly recommends migrating existing OpenShift clusters to etcd storage mode v3, particularly for larger clusters, or in scenarios where SSD storage is not available.
```

Comment:
I've updated the docs according to our discussions.  I put in an "IMPORTANT" box stating that future versions of OpenShift will require an etcd schema update.